src/hooks/useBoostedAPY.ts

```diff
@@ -0,0 +1,105 @@
+import { ProtocolAction } from '@aave/contract-helpers';
+import { ReserveIncentiveResponse } from '@aave/math-utils/dist/esm/formatters/incentive/calculate-reserve-incentives';
+
+import { ENABLE_SELF_CAMPAIGN, useMeritIncentives } from './useMeritIncentives';
+import { useMerklIncentives } from './useMerklIncentives';
+
+interface UseBoostedAPYProps {
+  symbol: string;
+  market: string;
+  protocolAction?: ProtocolAction;
+  protocolAPY: number;
+  incentives?: ReserveIncentiveResponse[];
+  address?: string;
+}
+
+interface BoostedAPYResult {
+  displayAPY: number | 'Infinity';
+  protocolIncentivesAPR: number | 'Infinity';
+  meritIncentivesAPR: number;
+  merklIncentivesAPR: number;
+  selfAPY: number;
+  totalMeritAPY: number;
+  hasInfiniteIncentives: boolean;
+  hasIncentives: boolean;
+  isBorrow: boolean;
+  breakdown: {
+    protocol: number;
+    protocolIncentives: number | 'Infinity';
+    merit: number;
+    merkl: number;
+    self: number;
+  };
+}
+
+export const useBoostedAPY = ({
+  symbol,
+  market,
+  protocolAction,
+  protocolAPY,
+  incentives = [],
+  address,
+}: UseBoostedAPYProps): BoostedAPYResult => {
+  // Calculate protocol incentives APR
+  const protocolIncentivesAPR = incentives.reduce((sum, inc) => {
+    if (inc.incentiveAPR === 'Infinity' || sum === 'Infinity') {
+      return 'Infinity';
+    }
+    return sum + +inc.incentiveAPR;
+  }, 0 as number | 'Infinity');
+
+  const { data: meritIncentives } = useMeritIncentives({
+    symbol,
+    market,
+    protocolAction,
+    protocolAPY,
+    protocolIncentives: incentives,
+  });
+
+  const { data: merklIncentives } = useMerklIncentives({
+    market,
+    rewardedAsset: address,
+    protocolAction,
+    protocolAPY,
+    protocolIncentives: incentives,
+  });
```

Comment:
proper way to handle which asset is passed? ie is it the underlying or is it the atoken?